drivers/usbdev/usbmsc_scsi.c

```diff
@@ -2454,9 +2470,34 @@ static int usbmsc_cmdwritestate(FAR struct usbmsc_dev_s *priv)
           priv->nsectbytes += nbytes;
           priv->nreqbytes  -= nbytes;
 
+#ifdef CONFIG_USBMSC_WRMULTIPLE
+          uint32_t nrbufs = MIN(priv->u.xfrlen, CONFIG_USBMSC_NWRREQS);
+
           /* Is the I/O buffer full? */
 
-          if (priv->nsectbytes >= lun->sectorsize)
+          if ((priv->nsectbytes >= lun->sectorsize * priv->u.xfrlen) ||
+              (priv->nsectbytes >= lun->sectorsize * CONFIG_USBMSC_NWRREQS))
+            {
+              /* Yes.. Write next sectors */
+
+              nwritten = USBMSC_DRVR_WRITE(lun, priv->iobuffer,
+                                           priv->sector, nrbufs);
+              if (nwritten < 0)
+                {
+                  usbtrace(TRACE_CLSERROR(USBMSC_TRACEERR_CMDWRITEWRITEFAIL),
+                           -nwritten);
+                  lun->sd     = SCSI_KCQME_WRITEFAULTAUTOREALLOCFAILED;
+                  lun->sdinfo = priv->sector;
+                  goto errout;
+                }
+
+              priv->nsectbytes = 0;
+              priv->residue   -= lun->sectorsize * nrbufs;
+              priv->u.xfrlen  -= nrbufs;
+              priv->sector    += nrbufs;
+            }
```

Comment:
```suggestion
            }
          else
```

Reply:
You mean get rid of the #else and put else instead?  It will not work then, condition if ((priv->nsectbytes >= lun->sectorsize)) is taken and it will not gather multiple packets?

Reply:
no, I mean this:
```
#ifdef ...
if (...)
  {
    ....
  }
else
#else
  if (...)
   {
     ...
   }
#endif

Reply:
```
--- a/drivers/usbdev/usbmsc_scsi.c
+++ b/drivers/usbdev/usbmsc_scsi.c
@@ -2496,6 +2496,7 @@ static int usbmsc_cmdwritestate(FAR struct usbmsc_dev_s *priv)
               priv->u.xfrlen  -= nrbufs;
               priv->sector    += nrbufs;
             }
+          else
 #else
           if ((priv->nsectbytes >= lun->sectorsize))
             {
```
will result in:

```
make[1]: Entering directory '/home/eenurkka/nuttx-upstream/scsi/drivers'
CC:  usbdev/usbmsc_scsi.c
usbdev/usbmsc_scsi.c: In function 'usbmsc_cmdwritestate':
usbdev/usbmsc_scsi.c:2522:9: error: expected expression before '}' token
 2522 |         }
      |         ^
make[1]: *** [Makefile:91: usbmsc_scsi.o] Error 1
```
Maybe I'm missing something?

Reply:
It's strange, could you update your change?

Reply:
If I update, then it won't compile. That 'else' would point to nowhere, eg. :

if (something)
{
 do something
}
else

Reply:
else should be part of the next if:
```
  else
#else
   if ((priv->nsectbytes >= lun->sectorsize))
#endif
```